token/client/src/token.rs

```diff
@@ -210,12 +221,29 @@ where
         &self.pubkey
     }
 
-    pub fn with_payer<S2: Signer>(&self, payer: S2) -> Token<T, S2> {
+    pub fn with_payer(&self, payer: Arc<dyn Signer>) -> Token<T> {
         Token {
             client: Arc::clone(&self.client),
             pubkey: self.pubkey,
             payer,
             program_id: self.program_id,
+            nonce_account: self.nonce_account,
+            nonce_authority: self.nonce_authority,
+            // XXX FIXME this is wrong but i dont understand the implications of clone on RwLock
+            // in fact i dont understand why we use write at all why not just Option<String> and clone
+            memo: Arc::new(RwLock::new(None)),
```

Comment:
actually i think this is the only issue i had left with this stuff. memo is thrown away here but needs to be retained, but im not sure what all the `write()` and `RwLock` stuff is for, is there a reason we dont want to just clone strings like normal?

Reply:
The idea is to do `with_memo` and send the transaction immediately after, so something like `token.with_memo("hiiii").create_mint(....)`, and that way it gets cleared out during the send, and it won't be there for your next transaction.

Since we're in an async context, we can't just use a normal `Option<String>`, as mentioned later